Sources/Hooks/Hook/UseState.swift

```diff
@@ -28,15 +28,17 @@ private struct StateHook<State>: Hook {
             get: {
                 coordinator.state.state
             },
-            set: { newState in
+            set: { newState, transaction in
                 assertMainThread()
 
                 guard !coordinator.state.isDisposed else {
                     return
                 }
 
-                coordinator.state.state = newState
-                coordinator.updateView()
+                withAnimation(transaction.animation) {
+                    coordinator.state.state = newState
+                    coordinator.updateView()
+                }
```

Comment:
Does it work more properly if we use `withTransaction` instead?

Reply:
Every day you learn something new! Didn't know there was such API, works fine & definitely feels more appropriate.